src/public/app/widgets/buttons/global_menu.js

```diff
@@ -333,7 +333,7 @@ export default class GlobalMenuWidget extends BasicWidget {
 
         const latestVersion = await this.fetchLatestVersion();
         this.updateAvailableWidget.updateVersionStatus(latestVersion);
-        this.$updateToLatestVersionButton.toggle(latestVersion > glob.triliumVersion);
+        this.$updateToLatestVersionButton.toggle((Number(latestVersion) > Number(glob.triliumVersion)) ?? false);
```

Comment:
Typecasting to number doesn't directly work. I believe we have to compare each version component (major, minor, patch) individually.

```js
> Number("1.2.11") > Number("1.2.10")
false
```

Reply:
We could implement another library [meant for exactly this](https://github.com/npm/node-semver) with:
```js
import * as semver from 'semver';

this.$updateToLatestVersionButton.toggle(semver.gt(latestVersion, glob.triliumVersion));
```

But I'm not sure if we want to add another lib.

Reply:
The problem with adding a new library is that we are here at the client level so it would involve a bit of effort to add it. I would personally only extract the version comparison algorithm, since I think it's not that difficult to implement.

1. Obtain major, minor, patch by splitting the version by "." with a max number of elements of 3.
2. If the major of $b > a$, then $b$ is newer.
3. If the minor of $b > a$, then $b$ is newer.
4. If the patch of $b > a$, then $b$ is newer.
5. Otherwise, $b$ is older or equal to $a$ (no popup needed).

Reply:
I think this is the function that you're looking for:
```js
function compareVersions(v1, v2) {
  const v1Parts = v1.split('.').map(Number);
  const v2Parts = v2.split('.').map(Number);

  for (let i = 0; i < Math.max(v1Parts.length, v2Parts.length); i++) {
    const part1 = v1Parts[i] || 0;
    const part2 = v2Parts[i] || 0;

    if (part1 > part2) return 1;
    if (part1 < part2) return -1;
  }
  return 0;
}
```

Let me know if you wanna toss this in @rom1dep, or if you'd prefer if I did it :)